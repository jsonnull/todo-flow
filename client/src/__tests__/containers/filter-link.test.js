// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

import FilterLink from '../../containers/filter-link';

const middleware = [];
const mockStore = configureStore(middleware);

describe('FilterLink', () => {
  let tree;
  it('renders without crashing', () => {
    const store = mockStore({});
    const component = renderer.create(
      <Provider store={store}><FilterLink /></Provider>
    );
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('sets the correct filter when clicked', () => {
    const store = mockStore({});
    const component = renderer.create(
      <Provider store={store}><FilterLink filter="SHOW_ACTIVE" /></Provider>
    );
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(store.getActions()).toEqual([]);
    tree.props.onClick({ preventDefault: () => {} });
    expect(store.getActions()).toEqual([
      {
        type: 'SET_VISIBILITY_FILTER',
        filter: 'SHOW_ACTIVE'
      }
    ]);
  });
});
