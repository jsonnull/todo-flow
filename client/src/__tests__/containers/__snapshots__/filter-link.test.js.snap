exports[`FilterLink renders without crashing 1`] = `<span />`;

exports[`FilterLink sets the correct filter when clicked 1`] = `
<a
  href="#"
  onClick={[Function]} />
`;
